dali/test/python/test_operator_readers_webdataset_corner.py

```diff
@@ -0,0 +1,378 @@
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, less_components_error)
+
+    def no_extensions_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"512 1\n0\n")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+            index_file.name,
+            ["jpg", "cls"],
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, no_extensions_error)
+
+    def corresponding_component_size_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"512 1\n0 jpg\n")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+            index_file.name,
+            ["jpg", "cls"],
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, corresponding_component_size_error)
+
+    def earlier_final_offset_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"512 2\n0 jpg 0\n1024 png 512")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+            index_file.name,
+            ["jpg", "cls"],
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, earlier_final_offset_error)
+
+    def offset_not_divisible_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"3072 2\n0 jpg 0\n1030 png 512")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/sample-tar/types_contents.tar",
+            index_file.name,
+            "float16",
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, offset_not_divisible_error)
+
+    def final_offset_not_divisible_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"3100 2\n0 jpg 0\n1024 png 512")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/sample-tar/types_contents.tar",
+            index_file.name,
+            "float16",
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, offset_not_divisible_error)
```

Comment:
Again, please check error message using nose_utils.

Reply:
done, will be pushed soon